exploration-module/src/main/java/io/edpn/backend/exploration/application/domain/System.java

```diff
@@ -1,25 +1,22 @@
 package io.edpn.backend.exploration.application.domain;
 
-import java.util.Objects;
-import java.util.UUID;
-
-public record System(
-        UUID id,
-        Long eliteId,
-        String name,
-        String starClass,
-        Coordinate coordinate
-) {
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
 
-    public System withEliteId(Long eliteId) {
-        return Objects.equals(this.eliteId, eliteId) ? this : new System(id, eliteId, name, starClass, coordinate);
-    }
-
-    public System withStarClass(String starClass) {
-        return Objects.equals(this.starClass, starClass) ? this : new System(id, eliteId, name, starClass, coordinate);
-    }
+import java.util.UUID;
 
-    public System withCoordinate(Coordinate coordinate) {
-        return Objects.equals(this.coordinate, coordinate) ? this : new System(id, eliteId, name, starClass, coordinate);
-    }
+@Getter
+@Setter
+@ToString
+@AllArgsConstructor
+@Builder
+public class System{
+    private UUID id;
+    private Long eliteId;
+    private String name;
+    private String starClass;
```

Comment:
Systems can have more than one star...

Reply:
System doesn't store List<Star> star class is just for the primary fuel scoop.

With the exception of rings all of our data is structured with child entities housing parent keys. Rather than parents having child lists. They're not bidirectional.

The star table will have the system_id if we need to collect the stars from a system.

Reply:
Updated to primaryStarClass for clarity